File: ts/wam/3656.ts
export const WamRevokeMessageSend = 3656 as const

export interface RevokeMessageSendEvent {
  /** field 1, wire `message_send_result_is_terminal` */
  messageSendResultIsTerminal?: boolean
  /** field 2, wire `message_type` */
  messageType?: MESSAGETYPE
  /** field 3, wire `resend_count` */
  resendCount?: number
  /** field 4, wire `retry_count` */
  retryCount?: number
  /** field 5, wire `revoke_duration` */
  revokeDuration?: number
  /** field 6, wire `revoke_type` */
  revokeType?: REVOKETYPE
}

File: go/wam/3656.go
const WamRevokeMessageSend = 3656 // channel: regular

type RevokeMessageSendEvent struct {
	MessageSendResultIsTerminal bool `wam:"message_send_result_is_terminal"` // field 1
	MessageType MESSAGETYPE `wam:"message_type"` // field 2
	ResendCount int64 `wam:"resend_count"` // field 3
	RetryCount int64 `wam:"retry_count"` // field 4
	RevokeDuration int64 `wam:"revoke_duration"` // field 5
	RevokeType REVOKETYPE `wam:"revoke_type"` // field 6
}